recipes/automated-testing.md

```diff
@@ -0,0 +1,85 @@
+# Automated Testing
+
+## Types of Automated Tests
+
+There are different approaches to testing, and depending on the level of the
+entry point, we can split tests into following categories.
+
+- **Unit Tests**
+- **Integration Tests**
+- **E2E Tests**
+- **Load/Performance Tests**
+- **Visual Tests**
+
+*Note that some people can call these tests by different names, but for Studion
+internal purposes, this should be considered the naming convention.*
+
+### Unit Tests
+
+These are the most isolated tests that we can write. They should take a specific
+function/service/helper/module and test its functionality. Unit tests will
+usually require mocked data, but since we're testing the case when specific
+input produces specific output, the mocked data set should be minimal.
+
+Unit testing is recommended for functions that contain a lot of logic and/or branching.
+It is convenient to test a specific function at the lowest level so if the logic
+changes, we can make minimal changes to the test suite and/or mocked data.
+
+
+### Integration Tests (API Tests)
+
+This is the broadest test category. With these tests, we want to make sure our
+API contract is valid and the API returns the expected data. That means we write
+tests for the publically available endpoints.
+
+**TODO**: do we want to add that we should run full backend for these type of tests?
```

Comment:
The important aspect is that we're not mocking the app's infrastructure dependencies, such as a database or other interactions with the "real world".
There's an exception when we want to isolate the boundaries of a test to avoid depending on the 3rd party services.

Also, API tests are just a subset of integration tests. In this section, it seems that Integration Tests == API Tests.
Integration tests can verify, for example, whether data is persisted as expected or not.

Reply:
I've split this into 2 sections: Integration and API tests based on your comment and Marko Lovric's feedback.